mart/attack/perturber.py

```diff
@@ -0,0 +1,117 @@
+#
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Callable
+
+import pytorch_lightning as pl
+import torch
+
+from .gradient_modifier import GradientModifier
+from .projector import Projector
+
+if TYPE_CHECKING:
+    from .initializer import Initializer
+    from .objective import Objective
+    from .threat_model import ThreatModel
+
+__all__ = ["LitPerturber"]
+
+
+class LitPerturber(pl.LightningModule):
```

Comment:
Why can't this be a `LitModular`? Then we inherit all of the optimizer functionality. The problem, I think, is the modules functionality. However, that should, in theory, cleanup the use of objective and gain?

Reply:
As I thought about this more, it would be really nice for this to be a LitModular because then this code could be used to generate attacks using the Lightning framework.

Reply:
I'll leave this for another PR.